src/rpc.rs

```diff
@@ -294,8 +290,11 @@ impl<T: Runtime> Rpc<T> {
         key: &StorageKey,
         hash: Option<T::Hash>,
     ) -> Result<Option<StorageData>, Error> {
-        let params = Params::Array(vec![to_json_value(key)?, to_json_value(hash)?]);
-        let data = self.client.request("state_getStorage", params).await?;
+        let params: &[_] = &[to_json_value(key)?, to_json_value(hash)?];
```

Comment:
needed to not be treated as an reference to an array i.e, a slice.

Reply:
It's annoying having to add this type hint, perhaps we could extract the whole pattern of: construct params, do request, then log. 

Anyway it's fine for now, hope in the future all these rpc calls can be generated with a macro. Is there an updated one in jsonrpsee client yet? When I first did this there was one but it didn't work.

Reply:
They can be generated by the macros already but the macros doesn't work when you have two different clients I think, then we don't from which crate the fetch types in each client.

The way it's designed is to use the wrapper `jsonrpsee crate` similar to what it was before but we don't expose the `tokio 02` feature there so can't be used here unfortunately.